addons/mail/models/mail_activity.py

```diff
@@ -572,53 +585,161 @@ def action_open_document(self):
     def activity_format(self):
         activities = self.read()
         self.mail_template_ids.fetch(['name'])
+        self.attachment_ids.fetch(['name'])
         for record, activity in zip(self, activities):
             activity['mail_template_ids'] = [
                 {'id': mail_template.id, 'name': mail_template.name}
                 for mail_template in record.mail_template_ids
             ]
+            activity['attachment_ids'] = [
+                {'id': attachment.id, 'name': attachment.name}
+                for attachment in record.attachment_ids
+            ]
         return activities
 
     @api.model
-    def get_activity_data(self, res_model, domain, limit=None, offset=0):
+    def get_activity_data(self, res_model, domain, limit=None, offset=0, fetch_done=False):
```

Comment:
After reading 'get_activity_data' twice, have to say it could benefits from some simplification / rewriting ... added various comments before this one, globally: try to use short (and meaningful) variables names, use recordsets, and... well... not sure but it seems to have a lot of dict(tuple(list(set)))) and we don't really know what lies inside the final variables ^^ . Would say this needs to be smoother.